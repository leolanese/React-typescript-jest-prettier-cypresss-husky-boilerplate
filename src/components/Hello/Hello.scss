.sample {
  align-items: center;
  color: rgb(156, 156, 156);
  display: flex;
  justify-content: center;
}
