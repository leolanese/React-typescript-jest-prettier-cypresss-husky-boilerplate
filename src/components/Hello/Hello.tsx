import * as React from 'react';

import style from './Hello.scss';

interface IHelloProps {
  foo: string;
  bar: string;
}

export const Hello = (props: IHelloProps): JSX.Element => (
  <div className={style.sample}>
    Hello from {props.foo} and {props.bar}!
  </div>
);
