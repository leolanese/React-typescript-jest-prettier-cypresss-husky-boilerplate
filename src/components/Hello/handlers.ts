export const initialComponentProps = {
  bar: 'Ipsum',
  foo: 'Lorem',
};

export const dummyFunc = (a: number, b: number): number => a + b;
