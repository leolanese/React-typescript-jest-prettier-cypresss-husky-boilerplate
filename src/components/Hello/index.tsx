import React, { useState } from 'react';

import { initialComponentProps } from '@components/Hello/handlers';
import { Hello } from '@components/Hello/Hello';

export default (): JSX.Element => {
  const [componentProps] = useState(initialComponentProps);

  return <Hello {...componentProps} />;
};
