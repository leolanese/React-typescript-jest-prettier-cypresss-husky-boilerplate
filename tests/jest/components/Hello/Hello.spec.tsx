import React from 'react';
import renderer from 'react-test-renderer';

import { Hello } from '@components/Hello/Hello';

describe('Hello component', () => {
  test('should render component properly', () => {
    // given
    const dummyComponentProps = {
      bar: 'test',
      foo: 'lorem',
    };

    // when
    const componentRenderer = renderer.create(<Hello {...dummyComponentProps} />);
    const tree = componentRenderer.toJSON();

    // then
    expect(tree).toMatchSnapshot();
  });
});
