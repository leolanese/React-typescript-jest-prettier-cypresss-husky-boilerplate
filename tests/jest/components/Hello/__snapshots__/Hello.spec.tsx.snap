// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello component should render component properly 1`] = `
<div
  className="sample"
>
  Hello from 
  lorem
   and 
  test
  !
</div>
`;
